RELEASE.md

```diff
@@ -1,4 +1,9 @@
 # Release Notes
+## [4.2.0] - 2024-07-09
```

Comment:
I see that we're bumping to `4.2` now. Does this have any backward incompatibility changes and hence the need to bump the minor?

Reply:
No, as this version adds an important feature (customer ask), we are bumping the minor version. We follow [semantic versioning](https://semver.org/).